### PR TITLE
Add "typecheck:stats" task to display how many files are typechecked

### PR DESCRIPTION
**Pull request by hsablonniere**
Display something like this:

image

We can always adjust the categories and patterns in the future...

This PR also does some cleanup in the tasks defined in `package.json` and documented in `tasks.md`. The rest of the documentation update will be done through #674.

### CODE REVIEW

File: tasks/generate-icons-assets.js [hunk 1]
```diff
@@ -27,9 +27,9 @@ async function run() {
  * @param {object} config config defining icons to be built
  * @param {string} config.namespace icon namespace
  * @param {string} config.glob icon files glob
- * @param {array<object>|null} config.svgoPlugins plugins to pass to SVGO
+ * @param {Array<object>|null} config.svgoPlugins plugins to pass to SVGO
```
Comment (pdesoyres-cc) [inline tasks/generate-icons-assets.js:30]: nitpick: maybe split changes in that file into a separate commit

File: tasks/typechecking-stats.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import rawGlob from 'glob';
+import json5 from 'json5';
+import fs from 'node:fs/promises';
+import { promisify } from 'node:util';
+
+const globOne = promisify(rawGlob);
+
+/**
+ * @param {string[]} patterns
```
Comment (pdesoyres-cc) [inline tasks/typechecking-stats.js:9]: nitpick: I prefer `Array<string>` instead of `string[]`. Do what you want with that.

**Review by pdesoyres-cc — CHANGES_REQUESTED** [review]
Nice. I have just a few comments.

**Review by florian-sanders-cc — APPROVED** [review]
Thanks for this command :heart_eyes_cat: 

I think it does a great good job and I like the categories you have selected :+1:

We may need to revisit it later on when we have many components included and we decide to rely on `exclude` and / or glob patterns.

**Review by HeleneAmouzou — APPROVED** [review]
Thanks and GG ! :clap: 
This PR made me discover new files in the `clever-components` project !

File: tasks/typechecking-stats.js (hunk 2 above)
Comment (hsablonniere) [reply tasks/typechecking-stats.js:9]: I have two wishes:

* that we (the JS world) did not have both similar options in the first place
* that we (the component maintainers) always did the same

I updated to `Array<string>` :wink:
